p4include/v1model.p4

```diff
@@ -117,7 +119,7 @@ enum MeterType {
     bytes
 }
 
-extern counter {
+extern counter<I> {
```

Comment:
I don't think we can break compatibility with existing programs; there are a lot of tutorials out there that may be using this file. I can see two ways forward:
- version the file; add a `const string v1model_version = "1.0.0";` and have the back-end do things depending on this version. We keep multiple v1model files around, and the user symlinks or uses -I to indicate the desired one
- deprecate the existing counter, the way we did it with the checksum apis, and add a new counter class which is generic. Old things will keep working, but we may duplicate some significant amount of code in the backend. The front-end should not really care.
We should version v1model anyway; the question is whether we add a new file or keep modifying things in the old one.
Maybe we can start with a PR with just the versioning. This would have been a great subject to discuss during the LDWG.